node_modules/browser/theme.js

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define((require, exports, module) => {
+
+// Note that all functions of `url` need to be called as methods.
+const url = require("./util/url.js");
+
+'use strict';
+
+const defaultTheme = {
+  glyphsShowing: false,
+  close: "#FC5753",
+  min: "#FDBC40",
+  max: "#33C748",
+  progress: "rgba(255, 255, 255, 0.4)",
+  location: "#eee",
+  buttons: "rgba(0, 0, 0, 0.5)",
+  domain: "rgba(0, 0, 0, 0.65)",
+  title: "rgba(0, 0, 0, 0.5)",
+  background: "#fff"
+};
+exports.defaultTheme = defaultTheme;
+
+const hardcodedColors = {
+  // [foreground, background]
+  "youtube.com": ["#cc181e", "#fff", false],
+  "yahoo.com": ["#2d1152", "#fff", false],
+  "facebook.com": ["#fff", "#3A5795", true],
+  "biadu.com": ["#2529d8", "#fff", false],
+  "amazon.com": ["#e47911", "#fff", false],
+  "taobao.com": ["#ff4400", "#fff", false],
+  "qq.com": ["#5da4e6", "#fff", false],
+  "sina.com.cn": ["#fff", "#ff8500", true],
+  "instagram.com": ["#fff", "#5380a5", true],
+  "imgur.com": ["#fff", "#2b2b2b", true],
+  "cnn.com": ["#fff", "#0c0c0c", true],
+  "slideshare.net": ["#fff", "#313131", true],
+  "deviantart.com": ["#fff", "#475c4d", true],
+  "soundcloud.com": ["#fff", "#383838", true],
+  "mashable.com": ["#fff", "#00aeef", true],
+  "daringfireball.net": ["#fff", "#4a525a", true],
+  "firewatchgame.com": ["#EF4338", "#2D102B", true],
+  "whatliesbelow.com": ["#fff", "#74888B", true],
+  "supertimeforce.com": ["#2EBCEC", "#051224", true]
+};
+
+const expandTheme = (foregroundColor, backgroundColor, isDark) => ({
+  isDark: isDark,
+  glyphsShowing: true,
+  close: foregroundColor,
+  min: foregroundColor,
+  max: foregroundColor,
+  progress: isDark ? "rgba(255, 255, 255, 0.2)" : "rgba(0, 0, 0, 0.2)",
```

Comment:
Let's talk about the progress bar. I think we can change the design just a bit to make it easier to color theme.